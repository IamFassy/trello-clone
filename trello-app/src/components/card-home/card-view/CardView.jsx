import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import IconButton from '../../../common/icon-button/IconButton';
import SingleCard from '../../../common/single-card/SingleCard';
import { addCard } from '../../../reducers/cardData';
import './card-view.scss';
import { library } from '@fortawesome/fontawesome-svg-core';
import { faPlus, faPen, faXmark } from '@fortawesome/free-solid-svg-icons';

library.add(faPlus, faPen, faXmark);

const CardView = () => {
	const cards = useSelector((state) => state.cardData.cards);
	const dispatch = useDispatch();
	const handleAddCard = () => {
		dispatch(addCard({ id: cards.length + 1 }));
	};

	return (
		<div className='card-view'>
			{cards.map((card) => {
				return (
					<SingleCard
						key={card.id}
						card={card}
						totalLength={cards.length}
					/>
				);
			})}
			<IconButton
				className='card-view__add-card'
				onClick={handleAddCard}
				testId={'add-card'}
			>
				<FontAwesomeIcon icon='fa-solid fa-plus' />
			</IconButton>
		</div>
	);
};

export default CardView;
