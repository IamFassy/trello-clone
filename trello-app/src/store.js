import { configureStore } from '@reduxjs/toolkit';
import cardDataReducer from './reducers/cardData';

export const store = configureStore({
	reducer: {
		cardData: cardDataReducer,
	},
});
