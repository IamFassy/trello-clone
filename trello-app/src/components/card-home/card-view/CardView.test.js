import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import CardView from './CardView';
import { Provider } from 'react-redux';
import { store } from '../../../store';
import reducer, { addCard } from '../../../reducers/cardData';

afterEach(cleanup);

test('should take a snapshot', () => {
	const { asFragment } = render(
		<Provider store={store}>
			<CardView />
		</Provider>
	);
	expect(
		asFragment(
			<Provider store={store}>
				<CardView />
			</Provider>
		)
	).toMatchSnapshot();
});

test('should add a new card', () => {
	const arr = {
		cards: [
			{
				id: 1,
				tasks: [],
				title: 'Change Title',
			},
			{
				id: 2,
				tasks: [],
				title: 'Change Title',
			},
		],
	};
	render(
		<Provider store={store}>
			<CardView />
		</Provider>
	);
	fireEvent.click(screen.getByTestId('add-card'));
	const previousState = {
		cards: [
			{
				id: 1,
				tasks: [],
				title: 'Change Title',
			},
		],
	};
	expect(reducer(previousState, addCard({ id: 2 }))).toEqual(arr);
});
