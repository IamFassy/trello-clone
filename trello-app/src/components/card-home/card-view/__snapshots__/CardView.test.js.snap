// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="card-view"
  >
    <div
      class="single-card"
    >
      <input
        class="single-card__title-holder"
        data-testid="title-name"
        value="Change Title"
      />
      <form
        class="add-task"
      >
        <div
          class="add-task__input-view"
        >
          <input
            data-testid="task-name"
            placeholder="Add Task"
            value=""
          />
        </div>
        <div
          class="add-task__btn-view"
        >
          <button
            class="icon-button "
            data-testid="add-task"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-plus "
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </form>
      <button
        class="single-card__delete-btn"
        data-testid="delete-card"
        disabled=""
      >
        Delete
      </button>
    </div>
    <button
      class="icon-button card-view__add-card"
      data-testid="add-card"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-plus "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432 256c0 17.69-14.33 32.01-32 32.01H256v144c0 17.69-14.33 31.99-32 31.99s-32-14.3-32-31.99v-144H48c-17.67 0-32-14.32-32-32.01s14.33-31.99 32-31.99H192v-144c0-17.69 14.33-32.01 32-32.01s32 14.32 32 32.01v144h144C417.7 224 432 238.3 432 256z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
